vault/seal.go

```diff
@@ -160,25 +160,42 @@ func (d *defaultSeal) BarrierConfig(ctx context.Context, ns *namespace.Namespace
 		return nil, err
 	}
 
-	view := NamespaceView(d.core.barrier, ns).SubView(barrierSealConfigPath)
+	view := d.core.NamespaceView(ns).SubView(barrierSealConfigPath)
 
-	// Fetch the core configuration
-	pe, err := d.core.physical.Get(ctx, view.Prefix())
-	if err != nil {
-		d.core.logger.Error("failed to read seal configuration", "error", err)
-		return nil, fmt.Errorf("failed to check seal configuration: %w", err)
-	}
+	var sealConfig []byte
 
-	// If the seal configuration is missing, we are not initialized
-	if pe == nil {
-		d.core.logger.Info("seal configuration missing, not initialized")
-		return nil, nil
+	if ns.IsRoot() {
+		// Fetch the core configuration
+		pe, err := d.core.physical.Get(ctx, view.Prefix())
+		if err != nil {
+			d.core.logger.Error("failed to read seal configuration", "error", err)
+			return nil, fmt.Errorf("failed to check seal configuration: %w", err)
+		}
+
+		// If the seal configuration is missing, we are not initialized
+		if pe == nil {
+			d.core.logger.Info("seal configuration missing, not initialized")
+			return nil, nil
+		}
+		sealConfig = pe.Value
+	} else {
+		se, err := view.Get(ctx, view.Prefix())
```

Comment:
view is already scoped to path (I think) so this should be
```suggestion
		se, err := view.Get(ctx, "")
```